trezorlib/tx_api.py

```diff
@@ -104,6 +109,30 @@ def json_to_tx(coin, data):
             extra_data_len = 1 + joinsplit_cnt * 1802 + 32 + 64
             t.extra_data = rawtx[-extra_data_len:]
 
+    if is_dash(coin):
+        t.type = data["type"]
+        if t.version == 3 and t.type != 1:
+            # It's a DIP2 special TX with payload
+            t.extra_data_len = data["extraPayloadSize"]
+            if t.extra_data_len == 0:
+                t.extra_data_len = 1
+                t.extra_data = b"\x00"
+            elif t.extra_data_len < 253:
+                t.extra_data = struct.pack("<B", t.extra_data_len)
+                t.extra_data_len += 1
+            elif t.extra_data_len <= 65535:
+                t.extra_data = struct.pack("<BH", 253, t.extra_data_len)
+                t.extra_data_len += 3
+            else:
+                t.extra_data = struct.pack("<BI", 254, t.extra_data_len)
```

Comment:
use `<BL` instead

also please add the `<BQ` case

Reply:
Done (in `pack_varint`)